sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -267,181 +271,34 @@ def plot_partial_dependence(
-            )
-
-    for i in chain.from_iterable(features):
-        if i >= len(feature_names):
-            raise ValueError(
-                "All entries of features must be less than "
-                "len(feature_names) = {0}, got {1}.".format(len(feature_names), i)
-            )
-
-    if isinstance(subsample, numbers.Integral):
-        if subsample <= 0:
-            raise ValueError(
-                f"When an integer, subsample={subsample} should be positive."
-            )
-    elif isinstance(subsample, numbers.Real):
-        if subsample <= 0 or subsample >= 1:
-            raise ValueError(
-                f"When a floating-point, subsample={subsample} should be in "
-                "the (0, 1) range."
-            )
-
-    # compute predictions and/or averaged predictions
-    pd_results = Parallel(n_jobs=n_jobs, verbose=verbose)(
-        delayed(partial_dependence)(
-            estimator,
-            X,
-            fxs,
-            response_method=response_method,
-            method=method,
-            grid_resolution=grid_resolution,
-            percentiles=percentiles,
-            kind=kind,
-        )
-        for fxs in features
-    )
-
-    # For multioutput regression, we can only check the validity of target
-    # now that we have the predictions.
-    # Also note: as multiclass-multioutput classifiers are not supported,
-    # multiclass and multioutput scenario are mutually exclusive. So there is
-    # no risk of overwriting target_idx here.
-    pd_result = pd_results[0]  # checking the first result is enough
-    n_tasks = (
-        pd_result.average.shape[0]
-        if kind == "average"
-        else pd_result.individual.shape[0]
-    )
-    if is_regressor(estimator) and n_tasks > 1:
-        if target is None:
-            raise ValueError("target must be specified for multi-output regressors")
-        if not 0 <= target <= n_tasks:
-            raise ValueError("target must be in [0, n_tasks], got {}.".format(target))
-        target_idx = target
-
-    # get global min and max average predictions of PD grouped by plot type
-    pdp_lim = {}
-    for pdp in pd_results:
-        values = pdp["values"]
-        preds = pdp.average if kind == "average" else pdp.individual
-        min_pd = preds[target_idx].min()
-        max_pd = preds[target_idx].max()
-        n_fx = len(values)
-        old_min_pd, old_max_pd = pdp_lim.get(n_fx, (min_pd, max_pd))
-        min_pd = min(min_pd, old_min_pd)
-        max_pd = max(max_pd, old_max_pd)
-        pdp_lim[n_fx] = (min_pd, max_pd)
-
-    deciles = {}
-    for fx in chain.from_iterable(features):
-        if fx not in deciles:
-            X_col = _safe_indexing(X, fx, axis=1)
-            deciles[fx] = mquantiles(X_col, prob=np.arange(0.1, 1.0, 0.1))
-
-    display = PartialDependenceDisplay(
-        pd_results=pd_results,
-        features=features,
+    return PartialDependenceDisplay.from_estimator(
```

Comment:
For the other classes, @NicolasHug was in favor to keep the function untouched and instead to have a bit of duplicated code during the deprecation.

Reply:
Okay, updated PR with the function body and the tests duplicated.